Vc/avx/const.h

```diff
@@ -119,8 +119,7 @@ namespace AVX
     {
 #ifdef Vc_IMPL_AVX2
 #if defined Vc_ICC || defined Vc_MSVC
-        __m256i allone;
-        allone = _mm256_cmpeq_epi8(allone, allone);
+        __m256i allone = _mm256_set1_epi64x(~0);
```

Comment:
Can we go with this for all compilers and remove the ifdefs? I think it should be optimal in all cases, no?

Reply:
Depends. I read somewhere that sometimes generating the constant is faster than reading a value from static memory. So we would need to check the generated code whether `~0` is stored in the static data or computed. But then, you are the expert on such matter, so I take your recommendation ;)

Reply:
It's computed for the other compilers, and I think that's actually ok, because at most that takes a few cycles, so it's like loading from L1. Loading from memory can be fast, or not, depending on where you're loading from.

Reply:
Let's use this implementation on all compilers now, but in a separate PR: #299.